backend/main.py

```
@@ -1244,6 +1346,50 @@ async def unload_model():
+    }
+
+    if model_name not in model_types:
+        raise HTTPException(status_code=400, detail=f"Unknown model: {model_name}")
+
+    model_type, model_size = model_types[model_name]
+
+    try:
+        if model_type == "tts":
+            tts_model = tts.get_tts_model()
+            if not tts_model.is_loaded() or getattr(tts_model, 'model_size', None) != model_size:
+                raise HTTPException(status_code=400, detail=f"Model {model_name} is not loaded")
```

Ruff (0.15.2)

[warning] Abstract `raise` to an inner function

(TRY301)